### PR TITLE
96000 part a eps service

### PR DESCRIPTION
**Pull request by lee-delarm6**
## Summary

- Added new EPS service configuration and base service classes.
- Updated settings.yml to include EPS service configurations.
- Introduced unit tests for the new EPS service and configuration classes.

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/96000

## Testing done

- New code is covered by unit tests.
- Old behavior: No EPS service existed prior to this change.
- Verification steps: Ensure all unit tests pass and API interactions through the EPS service work as expected.

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [x]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x]  Feature/bug has a monitor built into Datadog (if applicable)
- [x]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [x]  I added a screenshot of the developed feature

### CODE REVIEW

File: modules/vaos/app/services/eps/base_service.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module EPS
+  class BaseService < VAOS::SessionService
```
Comment (randomsync) [inline modules/vaos/app/services/eps/base_service.rb:4]: As we discussed, we're not gaining anything by inheriting from `VAOS::SessionService`. Rather we should just inherit from `Common::Client::Base`. We should also have a different statsd prefix to not confuse `'api.vaos'` metrics.

File: config/settings.yml [hunk 2]
```diff
@@ -2,7 +2,7 @@ hostname: 127.0.0.1:3000 # possible fallback for unsafe request.host
 
 vsp_environment: localhost
 
-virtual_hosts: ["127.0.0.1", "localhost"] # Safe host names
+virtual_hosts: [ "127.0.0.1", "localhost" ] # Safe host names
```
Comment (randomsync) [inline config/settings.yml:5]: would suggest avoiding the automatic IDE pretty-fying

**Comment by randomsync** [comment]
@lee-delarm6 not sure if I'm supposed to review this again yet, but your latest commit to this doesn't resolve the comments I posted earlier. also lint and other checks would need to be resolved.

File: modules/vaos/app/services/eps/base_service.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    attr_accessor :user
+
+    def initialize(user)
+      super(user)
+      @user = user
+    end
+
+    private
+
+    def perform(method, path, params, headers = nil, options = nil)
```
Comment (randomsync) [inline modules/vaos/app/services/eps/base_service.rb:16]: I'd suggest not making `perform` a private method

**Review by randomsync — CHANGES_REQUESTED** [review]

File: modules/vaos/app/services/eps/configuration.rb [hunk 4]
```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+module Eps
+  class Configuration < Common::Client::Configuration::REST
+    def self.login_url
```
Comment (randomsync) [inline modules/vaos/app/services/eps/configuration.rb:5]: we don't need to define class methods like this, and certainly don't need to define methods for every config setting. we can always do:
`delegate :base_path, :another_config, :yet_another, to: :settings`

File: modules/vaos/app/services/eps/base_service.rb [hunk 5]
```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+module Eps
+  class BaseService < Common::Client::Base
+    STS_OAUTH_TOKEN = :va_online_scheduling_sts_oauth_token
+
+    attr_accessor :user
```
Comment (RachalCassity) [inline modules/vaos/app/services/eps/base_service.rb:7]: Is  `attr_accessor :user `necessary? Would  `attr_reader :user `be more appropriate?

**Review by RachalCassity — CHANGES_REQUESTED** [review]

File: modules/vaos/spec/services/eps/base_service_spec.rb [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+  describe '#headers' do
+    it 'returns the correct headers' do
+      allow(RequestStore.store).to receive(:[]).with('request_id').and_return('request-id')
+
+      expected_headers = {
+        'Authorization' => 'Bearer 1234',
+        'Content-Type' => 'application/json',
+        'X-Request-ID' => 'request-id'
+      }
+      expect(service.send(:headers)).to eq(expected_headers)
```
Comment (randomsync) [inline modules/vaos/spec/services/eps/base_service_spec.rb:18]: `headers` is not private, suggest calling it directly, rather than `.send()`

**Review by randomsync — APPROVED** [review]
LGTM with minor changes.

File: modules/vaos/spec/services/eps/base_service_spec.rb (hunk 6 above)
Comment (lee-delarm6) [reply modules/vaos/spec/services/eps/base_service_spec.rb]: Gotcha, that makes sense

File: modules/vaos/app/services/eps/configuration.rb [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+        if ENV['VAOS_DEBUG'] && !Rails.env.production?
+          conn.request(:curl, ::Logger.new($stdout), :warn)
+          conn.response(:logger, ::Logger.new($stdout), bodies: true)
+        end
+
+        conn.response :betamocks if mock_enabled?
+        conn.response :snakecase
+        conn.response :json, content_type: /\bjson$/
+        conn.response :vaos_errors
+        conn.use :vaos_logging
```
Comment (RachalCassity) [inline modules/vaos/app/services/eps/configuration.rb:34]: Where is vaos_logging defined?

File: modules/vaos/app/services/eps/configuration.rb [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+
+        if ENV['VAOS_DEBUG'] && !Rails.env.production?
+          conn.request(:curl, ::Logger.new($stdout), :warn)
+          conn.response(:logger, ::Logger.new($stdout), bodies: true)
+        end
+
+        conn.response :betamocks if mock_enabled?
+        conn.response :snakecase
+        conn.response :json, content_type: /\bjson$/
+        conn.response :vaos_errors
```
Comment (RachalCassity) [inline modules/vaos/app/services/eps/configuration.rb:33]: I do not see vaos_logging anywhere else in the app. Is there a reason for this?

**Review by RachalCassity — APPROVED** [review]

**Review by randomsync — APPROVED** [review]

**Pull request merged**